### DISCUSSION TITLE
directory data file being ignored on --serve

### DISCUSSION THREAD

**Discussion by n8garrett** [11ty/eleventy#2821] [Uncategorized]
When working in dev mode using --serve, 11ty ignores my resources.json file inside the /resources/ directory. When I run a build command it works fine. Any idea why it doesn't work on --serve?

**Comment by zachleat** [comment]
are you using the vite plugin? I don’t think eleventy core changes behavior in serve/build in this way

**Reply by n8garrett** [reply]
Not using vite. It's a pretty basic set-up so far:

package.json
"scripts": {
    "clean": "rimraf _site",
    "dev:11ty": "eleventy --serve --watch",
    "build:11ty": "eleventy",
    "start": "run-p dev:*",
    "build": "run-s clean build:*"
  },
  "author": "Nate Garrett",
  "license": "ISC",
  "devDependencies": {
    "@11ty/eleventy": "^2.0.0",
    "npm-run-all": "^4.1.5",
    "rimraf": "^4.1.2"
  }

.eleventy.js
const dateFormatted = require('./src/_11ty/filters/dateFormatted');

module.exports = function(eleventyConfig) {
  eleventyConfig.addWatchTarget('./src/js/');
  eleventyConfig.addPassthroughCopy('./src/img');
  eleventyConfig.addPassthroughCopy('./src/js');
  
  // Filters
  eleventyConfig.addFilter('dateFormatted', dateFormatted);
  
  return {
    dir: {
      input: 'src',
      output: '_site'
    },
    markdownTemplateEngine: 'njk'
  }
}

**Reply by bobmonsour** [reply]
What do you mean by "works fine" when you run the build command? If /resources/ is at the root of your project, and not in /src, or in your _data directory, I'm not sure how it would use it. You could also consider adding the file as a watch target:

https://fns-demo-cloud--11ty.netlify.app/docs/watch-serve/#add-your-own-watch-targets

**Reply by n8garrett** [reply]
When I run npm run build it writes out all of the resource files correctly. 

When I run npm start it still writes all of the files correctly (same as the build), but if I add a new file to my resources directory is doesn't get run through the layout files and is not added to the resource tag (which I have defined in resources.json) when running with the --serve flag.

**Reply by n8garrett** [reply]
So in order to see my changes when I create a new resource file, I have to keep running npm start over and over. If I define the tag and template in the new-resource-created.md file's front matter, it works.

**Reply by bobmonsour** [reply]
It sounds like adding /resources/ as a watch target would make it work.

I'd suggest adding this to your eleventy config, depending on the location of your resources dir.

```
eleventyConfig.addWatchTarget("./src/resources/");
```

**Reply by n8garrett** [reply]
I gave that a try, but same results.

**Reply by bobmonsour** [reply]
Bummer. Can you share a repo to get a closer look?

**Reply by n8garrett** [reply]
https://github.com/n8garrett/web-docs

Appreciate you taking a look!

**Reply by bobmonsour** [reply]
When I run this command, it works for me...more below

```
npm run eleventy:serve
```

When the server is running, I've changed an existing md file in the resources dir and the site reflected the changes without restart.

AND, when I duplicated one of the existing files, it showed up without a restart. So, now I'm confused about what sequence of running eleventy is not working for you.

**Comment by n8garrett** [comment]
Thanks for testing, Bob.

I ran `npm run eleventy:serve` just now. When I created the first new file, it created it just fine (used the layout, tagged it, etc). When I created a 2nd new file, it created the file, but didn't use the layout or add it to the tag.

**Reply by bobmonsour** [reply]
Here's an image of what I get when I duplicated the image sizes (dup'd twice) and events file (once). It looks like your resources.json adds the tag "resources" to all of the files created with the md files. I don't see anywhere in your repo where you're trying to make use of the tags, other than iterating over the resources collection in category-pages.njk. Where do you expect to see the tags show up? What usage?
<img width="1156" alt="Screenshot 2023-02-22 at 1 21 14 PM" src="https://user-images.githubusercontent.com/133979/220762322-073a815c-bdf2-423c-97f7-d15ac7a1924d.png">

**Reply by n8garrett** [reply]
Looks like it's working as expected for you. However it not on my end. Could it be something with npm or node? Here's a screenshot showing what my vs code is doing after I duplicate 3 files.

Duplicate #1: image-sizes copy gets created in the resource directory with the layout applied.

Duplicate 2 and 3: image-sizes copy 2 and image-sizes copy 3 get created in the resources directory and as you can see in the code view, the only output is the content from the md file (nothing from the layout).


![resources](https://user-images.githubusercontent.com/9845114/220766387-f978b4de-2877-462c-af80-9f19ee948f9f.JPG)

**Reply by n8garrett** [reply]
A bit more info. It only does this for me when running 2.0.0 of 11ty. It works as expected on v1.0.0 and 1.0.1.

**Reply by n8garrett** [reply]
Maybe it's related to this issue https://github.com/11ty/eleventy/issues/2797?

**Reply by bobmonsour** [reply]
I did notice one other strange thing in your repo. You named the eleventy config file as simply

```
.eleventy
```

One thing that I did was change the name to

```
.eleventy.js
```

What you've used is not one of the accepted default filenames for the config file. 

See: https://www.11ty.dev/docs/config/#default-filenames

**Reply by bobmonsour** [reply]
What version of node are you using? 2.0 has a minimum requirement of node 14+. See https://www.11ty.dev/blog/eleventy-v2-beta/#breaking-changes for a list of all 2.0 breaking changes.

**Reply by n8garrett** [reply]
My node version is 18.13

And I have the file as .eleventy.js on my machine. The file didn't copy for some reason when I copied the repo from my work's GH to GH.com, so I created the file manually and misnamed it.

**Reply by bobmonsour** [reply]
Here's what I suggest. Download your repo into a new empty directory. Fix the .eleventy.js filename. Run these commands...and tell me if that works. I just did this exact set of things and it worked for me, adding new md files into the resources dir and they show up on the page.

```
npm init -y
npm install
npm run eleventy:serve
```

Let me know if that works. Fingers crossed.

**Reply by bobmonsour** [reply]
@n8garrett Any progress on this? Did what I suggest work?

**Reply by n8garrett** [reply]
Sometimes it works on 11ty 2.0 and sometimes it doesn't. It works consistently on 1.0.1 and below though. So I'm assuming it's related to https://github.com/11ty/eleventy/issues/2797.

**Reply by bobmonsour** [reply]
Sorry it isn't working for you. You might consider Adding a link to this discussion to that issue.
